processor/transformprocessor/internal/traces/traces.go

```diff
@@ -0,0 +1,368 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package traces // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/traces"
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/common"
+)
+
+// pathGetSetter is a getSetter which has been resolved using a path expression provided by a user.
+type pathGetSetter struct {
+	getter exprFunc
+	setter func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{})
+}
+
+func (path pathGetSetter) get(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{} {
+	return path.getter(span, il, resource)
+}
+
+func (path pathGetSetter) set(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{}) {
+	path.setter(span, il, resource, val)
+}
+
+func newGetSetter(val common.Value) (getSetter, error) {
+	if val.Path == nil {
+		return nil, fmt.Errorf("must be a trace path expression")
+	}
+
+	return newPathGetSetter(val.Path.Fields)
+}
+
+func newPathGetSetter(path []common.Field) (getSetter, error) {
+	var getter exprFunc
+	var setter func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{})
+	switch path[0].Name {
+	case "resource":
+		mapKey := path[1].MapKey
+		if mapKey == nil {
+			getter = func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{} {
+				return resource.Attributes()
+			}
+			setter = func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{}) {
+				if attrs, ok := val.(pdata.AttributeMap); ok {
+					resource.Attributes().Clear()
+					attrs.CopyTo(resource.Attributes())
+				}
+			}
+		} else {
+			getter = func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource) interface{} {
+				return getAttr(resource.Attributes(), *mapKey)
+			}
+			setter = func(span pdata.Span, il pdata.InstrumentationLibrary, resource pdata.Resource, val interface{}) {
+				setAttr(resource.Attributes(), *mapKey, val)
+			}
+		}
```

Comment:
Consider to extract this in a separate func with a suggestive name to improve readability.

Reply:
Please do for all the cases :)

Reply:
I think we will codegen this file in the future, and I suspect the codegen to look close to this. Is it helpful to split the functions still in the meantime or maybe make it easier to reason out the future codegen by being artificial codegen?

Reply:
For the moment is impossible to review, and I am not a codegen or codeparser (yet :) ), so I would prefer to be more readable until that change happens.

Reply:
Extracted funcs, thanks it looks much better